src/SSHLibrary/abstractclient.py

```diff
@@ -412,9 +414,8 @@ def read_until_regexp(self, regexp):
         :returns: The read output up and until the `regexp` matches.
         """
         regexp = self._encode(regexp)
-        if isinstance(regexp, basestring):
-            regexp = re.compile(regexp)
```

Comment:
This check cannot be removed. This method is documented (and hopefully also tested) to work also with already compiled regexp objects. Compilation should be done only if the given `regexp` is a string.